services_app/src/androidTest/java/org/opendatakit/activites/LoginActivity/BaseLoginActivity.java

```diff
@@ -0,0 +1,71 @@
+package org.opendatakit.activites.LoginActivity;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static com.google.common.truth.Truth.assertThat;
+
+import android.content.Context;
+import android.content.Intent;
+
+import androidx.test.core.app.ActivityScenario;
+import androidx.test.espresso.action.ViewActions;
+import androidx.test.espresso.intent.Intents;
+import androidx.test.espresso.matcher.RootMatchers;
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import org.junit.After;
+import org.junit.Before;
+import org.opendatakit.consts.IntentConsts;
+import org.opendatakit.properties.CommonToolProperties;
+import org.opendatakit.properties.PropertiesSingleton;
+import org.opendatakit.services.sync.actions.activities.LoginActivity;
+import org.opendatakit.services.sync.actions.fragments.UpdateServerSettingsFragment;
+
+import java.util.Map;
+
+public  abstract class BaseLoginActivity{
+    protected static final String TEST_SERVER_URL= "https://testUrl.com";
+    protected static ActivityScenario<LoginActivity> activityScenario;
+
+    @Before
+    public void setUp() {
+        String APP_NAME = "testAppName";
+
+        Intent intent = new Intent(getContext(), LoginActivity.class);
+        intent.putExtra(IntentConsts.INTENT_KEY_APP_NAME, APP_NAME);
+        activityScenario= ActivityScenario.launch(intent);
+        onView(withId(android.R.id.button2)).inRoot(RootMatchers.isDialog()).perform(ViewActions.click());
+        activityScenario.onActivity(activity -> {
+            PropertiesSingleton props = activity.getProps();
+            assertThat(props).isNotNull();
+
+            Map<String, String> serverProperties = UpdateServerSettingsFragment.getUpdateUrlProperties(TEST_SERVER_URL);
+            assertThat(serverProperties).isNotNull();
+            props.setProperties(serverProperties);
+
+            activity.updateViewModelWithProps();
+        });
+        Intents.init();
+    }
+    @After
+    public void clearTestEnvironment() {
+        activityScenario.onActivity(activity -> {
+            PropertiesSingleton props = activity.getProps();
+            assertThat(props).isNotNull();
+
+            Map<String, String> serverProperties = UpdateServerSettingsFragment.getUpdateUrlProperties(
+                    activity.getString(org.opendatakit.androidlibrary.R.string.default_sync_server_url)
+            );
+            assertThat(serverProperties).isNotNull();
+            serverProperties.put(CommonToolProperties.KEY_FIRST_LAUNCH,"true");
```

Comment:
I am not sure why KEY_FIRST_LAUNCH set to true should be a clean for the test environment. We can discuss in our meeting.